File: customer-manager-frontend/src/models/DynamicTableMetadata.ts
// DynamicTableMetadata.ts
export interface DynamicTableMetadata {
  id?: number;
  tableKey: string;        // 表唯一标识
  fieldName: string;       // 字段名
  fieldLabel: string;      // 显示标签
  fieldType: string;       // 字段类型(text/number/select/date等)
  placeholder?: string;    // 占位提示
  required: boolean;       // 是否必填
  regex?: string;          // 验证正则表达式
  options?: string;        // 下拉框选项(JSON格式)
  sortOrder?: number;      // 显示顺序
  defaultValue?: string;   // 默认值
  maxLength?: number;      // 最大长度限制
  min?: number;   // 添加这一行
  max?: number;   // 可一并添加
  decimalPlaces?: number;
}

File: customer-manager-backend/src/main/java/com/wushu/customer/manager/model/DynamicTableMetadata.java
package com.wushu.customer.manager.model;

import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class DynamicTableMetadata {
    private Long id;
    private String tableKey;        // 表唯一标识
    private String fieldName;       // 字段名
    private String fieldLabel;      // 显示标签
    private String fieldType;       // 字段类型(text/number/select/date等)
    private String placeholder;     // 占位提示
    private boolean required;       // 是否必填
    private String regex;           // 验证正则表达式
    private String options;         // 下拉框选项(JSON格式)
    private Integer sortOrder;      // 显示顺序
    private String defaultValue;    // 默认值
}